[Test Fix] Update ip addresses for ipv6 check

ColeHiggins2 opened the pull request:
tests are failing in ipv6 because we do not have checks for network type. 
Updating code to add a check for network type


```tests/foreman/ui/test_rhcloud_inventory.py:165: in test_rhcloud_inventory_e2e
    host['system_profile']['network_interfaces'][0]['ipv4_addresses'][0]
E   IndexError: list index out of range```

## Summary by Sourcery

Update RH Cloud inventory UI and API tests to handle both IPv4 and IPv6 network types and align expectations with current obfuscation behavior.

Documentation:
- Clarify test documentation to refer to generic IP address obfuscation covering both IPv4 and IPv6.

Tests:
- Adjust inventory tests to select IPv4 or IPv6 addresses based on the system network type when validating reported IPs.
- Update obfuscation tests to assert the correct presence or absence of hostnames, IP addresses, and installed packages in generated reports in line with current settings behavior.

ColeHiggins2 committed:
update ip address for ipv6 check

ColeHiggins2 commented:
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_inventory.py -k "test_rh_cloud_inventory_settings"

ColeHiggins2 reviewed (commented):
test passes in prt. not sure why its not showing here
```15:30:10  ========= 8 passed, 40 deselected, 348 warnings in 6353.08s (1:45:53) ==========```

jnagare-redhat reviewed (approved):
Ack LGTM

LadislavVasina1 commented on tests/foreman/api/test_rhcloud_inventory.py:
```diff
@@ -90,15 +91,28 @@ def test_rhcloud_inventory_api_e2e(
+        assert virtual_host.ip_addr in ip_addresses
+        assert baremetal_host.ip_addr in ip_addresses
+        assert virtual_host.ip_addr in ipv4_addresses
+        assert baremetal_host.ip_addr in ipv4_addresses
```
@ColeHiggins2 Could you please try it this way, I find it more compact. Should work too imo:
```suggestion
		# Dynamic key determination based on the network type
		is_ipv6 = module_target_sat.network_type == NetworkType.IPV6
		key = 'ipv6_addresses' if is_ipv6 else 'ipv4_addresses'
		
		ip_addresses = {
		    host['system_profile']['network_interfaces'][0][key][0]
		    for host in json_data['hosts']
		}
		
		ips = {host['ip_addresses'][0] for host in json_data['hosts']}
		
		for host_ip in (virtual_host.ip_addr, baremetal_host.ip_addr):
		    assert host_ip in ip_addresses
		    assert host_ip in ips
```

ColeHiggins2 committed:
condensing ip address check

ColeHiggins2 committed:
pre-commit

synkd reviewed (approved):
Looks like something is off in the logic of the new version, as all tests are now failing PRT with assertions on mismatched IP addresses.

However, I'm ACKing as Cole's original version passed PRT. @ColeHiggins2 I will leave it up to you whether to fix the more concise version or revert to your original version.

ogajduse reviewed (changes requested):
`ContentHost.ip_addr` does not work reliably and currently only returns the IPv4 address. Therefore, the test cannot rely on it to return an IPv6 address.

```python
@property
def ip_addr(self):
    ipv4, *ipv6 = self.execute('hostname -I').stdout.split()
    return ipv4
```

```python
[ins] In [1]: hostname_i_output = "10.202.145.224 10.40.2.190 2620:52:0:2802:5632:7419:bb83:80b2"

[ins] In [2]: ipv4, *ipv6 = hostname_i_output.split()

[ins] In [3]: ipv4
Out[3]: '10.202.145.224'

[ins] In [4]: ipv6
Out[4]: ['10.40.2.190', '2620:52:0:2802:5632:7419:bb83:80b2']
```

ColeHiggins2 committed:
fix failing prt

ColeHiggins2 commented:
trigger: test-robottelo
pytest: tests/foreman/ui/test_rhcloud_inventory.py -k "test_rh_cloud_inventory_settings"

ColeHiggins2 commented:
> `ContentHost.ip_addr` does not work reliably and currently only returns the IPv4 address. Therefore, the test cannot rely on it to return an IPv6 address.
> 
> ```python
> @property
> def ip_addr(self):
>     ipv4, *ipv6 = self.execute('hostname -I').stdout.split()
>     return ipv4
> ```
> 
> ```python
> [ins] In [1]: hostname_i_output = "10.202.145.224 10.40.2.190 2620:52:0:2802:5632:7419:bb83:80b2"
> 
> [ins] In [2]: ipv4, *ipv6 = hostname_i_output.split()
> 
> [ins] In [3]: ipv4
> Out[3]: '10.202.145.224'
> 
> [ins] In [4]: ipv6
> Out[4]: ['10.40.2.190', '2620:52:0:2802:5632:7419:bb83:80b2']
> ```

What do you recommend I do then? @ogajduse

ColeHiggins2 committed:
remove useless ip check

ColeHiggins2 commented:
trigger: test-robottelo
pytest: tests/foreman/api/test_rhcloud_inventory.py -k "test_rhcloud_inventory_api_e2e"
network_type: ipv6

ColeHiggins2 commented:
============================= test session starts ==============================
collected 1 item

tests/foreman/ui/test_rhcloud_inventory.py .                             [100%]

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================== 1 passed, 29 warnings in 615.31s (0:10:15) ==================
Finished running tests!

ColeHiggins2 commented:
PRT seems to be broken on this pr. but I manually testec all tests locally and they are passing.

LadislavVasina1 reviewed (approved):
ACK, let's see it in automation, but please keep an eye on it @ColeHiggins2 when it runs.
